File: tools/screenshot_ui_flow.py
import sys
import os
from playwright.sync_api import sync_playwright

URL = "http://localhost:8085/index.html"
VIEWPORT = {"width": 1280, "height": 720}
OUT_DIR = "screenshots"

os.makedirs(OUT_DIR, exist_ok=True)


def click_and_shot(page, x, y, name, wait=3000):
    page.mouse.click(x, y)
    page.wait_for_timeout(wait)
    path = os.path.join(OUT_DIR, f"{name}.png")
    page.screenshot(path=path)
    print(f"Saved {path}")


def main():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(viewport=VIEWPORT)
        page = context.new_page()
        page.goto(URL)
        page.wait_for_timeout(8000)
        page.screenshot(path=os.path.join(OUT_DIR, "01_title.png"))

        # Click "开始新游戏" button in center
        click_and_shot(page, 640, 420, "02_after_start", wait=5000)

        # Bottom navigation buttons approximate positions for 1280x720
        # They are in a HFlowContainer at bottom center, around y=640-680
        buttons = [
            ("battle_boss", 430, 660),
            ("equipment", 540, 660),
            ("shop", 640, 660),
            ("stats", 740, 660),
            ("achievements", 840, 660),
            ("quests", 940, 660),
        ]

        for name, x, y in buttons:
            click_and_shot(page, x, y, f"03_{name}", wait=3000)
            # close any opened sub-UI by clicking a close/return area if needed
            # For now, assume buttons toggle; click same to close if stays open

        browser.close()


if __name__ == "__main__":
    main()

File: tools/screenshot_ui_flow.js
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

const URL = 'http://localhost:8085/index.html';
const VIEWPORT = { width: 1280, height: 720 };
const OUT_DIR = 'screenshots';

if (!fs.existsSync(OUT_DIR)) fs.mkdirSync(OUT_DIR, { recursive: true });

async function clickAndShot(page, x, y, name, wait = 3000) {
    await page.mouse.click(x, y);
    await page.waitForTimeout(wait);
    const file = path.join(OUT_DIR, `${name}.png`);
    await page.screenshot({ path: file });
    console.log(`Saved ${file}`);
}

(async () => {
    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext({ viewport: VIEWPORT });
    const page = await context.newPage();
    await page.goto(URL);
    await page.waitForTimeout(8000);
    await page.screenshot({ path: path.join(OUT_DIR, '01_title.png') });

    await clickAndShot(page, 640, 420, '02_after_start', 5000);

    const buttons = [
        ['battle_boss', 110, 690],
        ['equipment', 320, 690],
        ['shop', 535, 690],
        ['stats', 750, 690],
        ['achievements', 960, 690],
        ['quests', 1175, 690],
    ];

    for (const [name, x, y] of buttons) {
        await clickAndShot(page, x, y, `03_${name}`, 3000);
        // 点击左上角返回按钮回到战斗主界面
        await page.mouse.click(45, 45);
        await page.waitForTimeout(1500);
    }

    await browser.close();
})();
